inst/report/scripts/generate_report.R

```diff
@@ -0,0 +1,705 @@
+  # Check if CARD data exists
+  if (!dir.exists(destination_dir)) {
+    dir.create(destination_dir)
+  }
+  if (!file.exists("CARD_data/aro_index.tsv")) {
+    # Step 1: Download CARD Data
+    download.file("https://card.mcmaster.ca/latest/data", "card_data.tar.bz2")
+    #unzip("card_data.tar.bz2", exdir = "CARD_data")
+    system(paste("tar -xjf card_data.tar.bz2 -C", destination_dir))
+  }
+
+  # Step 2: Open ARO_index.tsv
+  aro_data <- read.delim("CARD_data/aro_index.tsv", sep = "\t", header = TRUE)
+
+  # Step 3: Map CARD Short Name
+  antibiotics <- read.delim("CARD_data/shortname_antibiotics.tsv",
+                            sep = "\t", header = TRUE)
+  pathogens <- read.delim("CARD_data/shortname_pathogens.tsv",
+                          sep = "\t", header = TRUE)
+
+  aro_data <- aro_data %>%
+    mutate(
+      pathogen = str_extract(CARD.Short.Name, "^[^_]+"),
+      gene = str_extract(CARD.Short.Name, "(?<=_)[^_]+"),
+      drug = str_extract(CARD.Short.Name, "(?<=_)[^_]+$")
+    ) %>%
+    left_join(pathogens, by = c("pathogen" = "Abbreviation")) %>%
+    left_join(antibiotics, by = c("drug" = "AAC.Abbreviation"))
+
+  # Sort names
+  aro_data <- aro_data %>%
+    arrange(Pathogen, Molecule) %>%
+    group_by(Pathogen, Molecule)
+
+  # Filter data based on user input
+  filtered_data <- aro_data
+
+  if (!is.null(pathogen)) {
+    filtered_data <- filtered_data %>% filter(Pathogen == !!pathogen)
+  }
+
+  if (!is.null(drug)) {
+    filtered_data <- filtered_data %>% filter(Molecule == !!drug)
+  }
+
+  # Check if filtered data is empty
+  if (nrow(filtered_data) == 0) {
+    stop("No data found for the specified pathogen and drug.")
+  }
+
+  # Extract protein accessions
+  accessions <- filtered_data$Protein.Accession
+
+  # Function to fetch FASTA sequence from NCBI
+  get_fasta <- function(accession) {
+    entrez_fetch(db = "protein", id = accession, rettype = "fasta")
+  }
+
+  # Download sequences
+  fasta_sequences <- lapply(accessions, get_fasta)
+
+  # Write sequences to a file
+  writeLines(unlist(fasta_sequences), "filtered_proteins.fasta")
+
+  return("FASTA sequences downloaded to 'filtered_proteins.fasta'.")
+}
+
+# Run analysis
+#' @export
+run_analysis <- function(
+        upload_type = "Fasta",
+        evalue = 0.00001,
+        accnum_fasta_input = "",
+        file_paths = list(accnum = NULL, fasta = tempfile(), msa = tempfile(),
+                          blast = NULL, iprscan = NULL),
+        blast_db = "refseq_protein",
+        blast_hits = 100,
+        blast_eval = 0.0001,
+        acc_homology_analysis = TRUE,
+        acc_da_analysis = TRUE,
+        acc_phylogeny_analysis = FALSE,
+        report_template_path = "/report/report_template.Rmd",
+        output_file = file.path(tempdir(), "report.html"),
+        DASelect = "All",
+        mainSelect = NULL,
+        PhyloSelect = "All",
+        q_heatmap_select = "All",
+        DACutoff = 95,
+        GCCutoff = 0.5,
```

Comment:
GC (aka GenomicContext) code not in use currently.
```suggestion
        # GCCutoff = 0.5,
```